### PR TITLE
Add storage challenge to k8s track

### PR DESCRIPTION
**Pull request by martijnvg**
Add a new challenge to the k8s tsdb track to allow to better benchmark storage usage.
This challenge doesn't run any queries and just indexes the data and runs force merge at the end (for consistent storage results). Currently there are no plans to include this challenge with nightly. Just for adhoc runs.

### CODE REVIEW

**Comment by martijnvg** [comment]
@elasticmachine update branch

**Comment by gareth-ellis** [comment]
@elasticmachine update branch

**Comment by martijnvg** [comment]
@elasticmachine update branch

File: tsdb_k8s_queries/challenges/default.json [hunk 1]
```diff
@@ -326,4 +326,71 @@
...
+      "schedule": [
+        {
+          "name": "create-all-templates",
+          "operation": {
+            "operation-type": "create-composable-template",
+            "request-params": {
+              "create": "true"
+            }
+          }
+        },
```
Comment (gbanasiak) [inline tsdb_k8s_queries/challenges/default.json:343]: I think we are missing `delete-all-datastreams`, `delete-all-composable-template`, and `put-timestamp-template` tasks here, similarly to what is done in other challenges in this track. Challenge should restore a known good state before continuing with whatever it's supposed to do.

File: tsdb_k8s_queries/challenges/default.json [hunk 2]
```diff
@@ -326,4 +326,71 @@
         }{{ ", " if not loop.last else "" }}
         {%- endfor %}
       ]
-    }
+    },
+    {
+      "name": "storage",
```
Comment (gbanasiak) [inline tsdb_k8s_queries/challenges/default.json:331]: Cosmetic remark: conventional name here would be `append-no-conflicts-index-only`.

**Review by gbanasiak — APPROVED** [review]
LGTM. Serverless IT passed for the added challenge but failed for unrelated reasons, I've restarted it.

**Comment by martijnvg** [comment]
> Serverless IT passed for the added challenge but failed for unrelated reasons, I've restarted it.

Thanks!

**Pull request merged**